lib/scsi_defs.h

```diff
@@ -0,0 +1,121 @@
+#ifndef LIB_SCSI_DEFS_H
+#define LIB_SCSI_DEFS_H
+
+namespace scsi_defs {
+
+  enum Status {
```

Comment:
Could these enums all be `enum class`? Such enums, called "strongly typed" or "scoped" enums, tend to be less error-prone as they don't have implicit conversions to/from integers (though `static_cast` can be used for explicit conversion) and they clutter the namespace less.

Reply:
yep that's a good idea!